### DISCUSSION TITLE
Environment variable to disable idle timeout bypass permission

### DISCUSSION THREAD

**Discussion by JulianVennen** [PurpurMC/Purpur#1304] [__Completed__] [answered]
Hi, we noticed that purpur adds a permission which allows players to bypass the idle timeout:
https://github.com/PurpurMC/Purpur/blob/ver/1.19.4/patches/server/0012-AFK-API.patch#L91

This creates a problem for us at Aternos since we want to ensure that our limited resources are used for servers with active players on them.

I want to request an option to disable this feature/permission, ideally as an environment variable.

**Comment by Kichura** [comment]
Can't you just make a non-removable plugin instead for every aternos server that has purpur as it's software active that basically automates the shutdown event from bukkit instead of having to auto-run the idle timeout system every single time? *(unless it isn't possible to do that then perhaps an explanation could be shared there if needed)*

+ If people ever tried to use PlugMan to kill off the plugin then it is possible to have it not disable your own one if modified.

**Comment by encode42** [comment, marked as answer]
Added the `PURPUR_FORCE_IDLE_KICK` environment variable in adc78043f5f19537fcd0c271576e097065467911. (build #1955)

(may be removed in the future; debate is ongoing on the Discord server)

**Reply by granny** [reply]
Left a comment on my view on the matter so it's more accessible.

**Reply by JulianVennen** [reply]
Hi, this environment variable does not seem to prevent the permission I mentioned above.
If a player has the bypassIdleKick permission, then they still aren't kicked in https://github.com/PurpurMC/Purpur/blob/ver/1.19.4/patches/server/0012-AFK-API.patch#L106

Your implementation seems to only use this as an override to the config option, which does not solve the problem for us.

**Reply by granny** [reply]
> Hi, this environment variable does not seem to prevent the permission I mentioned above. If a player has the bypassIdleKick permission, then they still aren't kicked in https://github.com/PurpurMC/Purpur/blob/ver/1.19.4/patches/server/0012-AFK-API.patch#L106
> 
> Your implementation seems to only use this as an override to the config option, which does not solve the problem for us.

@JulianVennen In effect as of https://github.com/PurpurMC/Purpur/commit/9429820ef674811674c43475f55ed8a1cd76ed97 (build #1959).

If you will indulge me, I'm curious to know what the plan is for the older versions.

**Reply by JulianVennen** [reply]
Thank you!

Our current plan is to just ignore older versions, as they aren't played as much either way.

We already have other systems to prevent inactive players (e.g. people using auto clickers) they just take much longer to kick in.

**Comment by granny** [comment]
I don't see an issue with it as long as it ends up being used. (It's already in effect since @Encode42 added it).

Imo it's kind of pointless if you want to continue listing the latest purpur builds for each minecraft version (like you currently do). You still need to think of a solution for those versions, or just remove them as potential options.

But if you *do* end up finding a solution for all the other versions, at that point you might as well apply that same solution for 1.19.4, which would mean the env variable goes unused.

**Reply by granny** [reply]
I'd like to note in case it's brought up; We will *not* go through the effort of applying that change to every version of minecraft that Purpur is available in.
